src/nc/core/likec/Simplifier.cpp

```diff
@@ -346,6 +346,25 @@ std::unique_ptr<Expression> Simplifier::simplify(std::unique_ptr<BinaryOperator>
      */
     if (node->operatorKind() == BinaryOperator::ASSIGN) {
         if (VariableIdentifier *leftIdent = node->left()->as<VariableIdentifier>()) {
+             /*
+             * if the right hand side is a cast to the type of
+             * the left hand side, check to see if the right hand side expression
+             * will automatically be upgraded to the type of the left hand side
+             * only works for integer types.
+             */
```

Comment:
It makes sense to do this transformation for any kind of assignment, not only for an assignment to a variable.
Could you make your rewrite rule more generic?